### DISCUSSION TITLE
Fprime logo for the readme

### DISCUSSION THREAD

**Discussion by Anirban166** [nasa/fprime#1641] [Polls]
I was wondering if it would be good to have the Fprime logo in the readme - definitely not crucial to have one, it's just that I think it might look a bit more appealing than what we have currently: 

# F´: A Flight-Proven, Multi-Platform, Open-Source Flight Software Framework

[![Language grade: C++](https://img.shields.io/lgtm/grade/cpp/g/nasa/fprime.svg?logo=lgtm&logoWidth=18)](https://lgtm.com/projects/g/nasa/fprime/context:cpp) [![Language grade: Python](https://img.shields.io/lgtm/grade/python/g/nasa/fprime.svg?logo=lgtm&logoWidth=18)](https://lgtm.com/projects/g/nasa/fprime/context:python)

...
<hr>

And if it would be more appropriate to have some wording, then maybe some text that describes the project to go with the logo?

I like the current logo design of F' (it looks pretty cool to me!) and I definitely wouldn't change that but I think the readme could do with one. Also, I guess the font for 'A Flight Software and Embedded Systems Framework' could do with a better one for the banner image on the [website](https://nasa.github.io/fprime/) (like maybe one that has a relatively similar font style with respect to the logo or the letter 'F')

For example, I made some simple ones during my free time to illustrate that and here's what they *could* (just my take on it!) look like for the header portion of the readme:

For initial thought, I made a vectorized version of the F' logo: (i.e. a .svg, no changes in the design)

<p align = "center">
<img width = "150" height = "auto" src = "https://user-images.githubusercontent.com/30123691/186663338-8da68c10-dc9b-4948-88c5-7a12146eb270.png"> <br>   
<strong>A Flight-Proven, Multi-Platform, Open-Source Flight Software Framework</strong> <br>
</p>  
<p align = "center">
<img src = "https://img.shields.io/lgtm/grade/python/g/nasa/fprime.svg?logo=lgtm&logoWidth=18" alt = "Language grade: Python"> <img src = "https://img.shields.io/lgtm/grade/cpp/g/nasa/fprime.svg?logo=lgtm&logoWidth=18" alt = "Language grade: C++"> 
<a href = "https://lgtm.com/projects/g/nasa/fprime/context:python"></a> <br>   
</p>

...
<hr>

I then added a gradient of brown to represent Martian texture: (Ingenuity being the prime use case representation for fprime :)

<p align = "center">
<img width = "150" height = "auto" src = "https://user-images.githubusercontent.com/30123691/186662235-5ecb9402-7f21-4e67-9d9a-576fa8779795.png"> <br>   
<strong>A Flight-Proven, Multi-Platform, Open-Source Flight Software Framework</strong> <br>
</p>  
<p align = "center">
<img src = "https://img.shields.io/lgtm/grade/python/g/nasa/fprime.svg?logo=lgtm&logoWidth=18" alt = "Language grade: Python"> <img src = "https://img.shields.io/lgtm/grade/cpp/g/nasa/fprime.svg?logo=lgtm&logoWidth=18" alt = "Language grade: C++"> 
<a href = "https://lgtm.com/projects/g/nasa/fprime/context:python"></a> <br>   
</p>

...
<hr>

For reference to what I said regarding adding some wording to the logo, here's a version where I added the words from the banner image to the logo with a slightly better font:

<p align = "center">
<img width = "100%" height = "auto" src = "https://user-images.githubusercontent.com/30123691/186647333-c7c3bd66-bbe6-4111-aabb-e03b9893b243.png">
</p>
<p align = "center">
<strong>A Flight-Proven, Multi-Platform, Open-Source Flight Software Framework</strong>
</p>
<p align = "center">
<a href = "https://lgtm.com/projects/g/nasa/fprime/context:cpp"></a>
<img src = "https://img.shields.io/lgtm/grade/cpp/g/nasa/fprime.svg?logo=lgtm&logoWidth=18" alt = "Language grade: C++"> 
<a href = "https://lgtm.com/projects/g/nasa/fprime/context:python"></a>
<img src = "https://img.shields.io/lgtm/grade/python/g/nasa/fprime.svg?logo=lgtm&logoWidth=18" alt = "Language grade: Python"> 
</p>

...
<hr>

And this is the closest I got to in matching the font style:

<p align = "center">
<img width = "100%" height = "auto" src = "https://user-images.githubusercontent.com/30123691/186647775-4e166cca-c769-46b1-95a6-ab2374cb6409.png">
</p>
<p align = "center">
<strong>A Flight-Proven, Multi-Platform, Open-Source Flight Software Framework</strong>
</p>
<p align = "center">
<a href = "https://lgtm.com/projects/g/nasa/fprime/context:cpp"></a>
<img src = "https://img.shields.io/lgtm/grade/cpp/g/nasa/fprime.svg?logo=lgtm&logoWidth=18" alt = "Language grade: C++"> 
<a href = "https://lgtm.com/projects/g/nasa/fprime/context:python"></a>
<img src = "https://img.shields.io/lgtm/grade/python/g/nasa/fprime.svg?logo=lgtm&logoWidth=18" alt = "Language grade: Python"> 
</p>

...
<hr>

Made this a poll so the community can vote and decide if any of this would be good to have!

**Comment by astroesteban** [comment]
This is great! Thanks for putting these examples together 🚀 
I like the second one a lot. It looks awesome.

**Comment by ThibFrgsGmz** [comment]
Wonderful work!

The first two could be integrated into the GDS, the first in dark mode and the second in light mode.

Regarding the ones with the wording, I prefer the second one which looks more futuristic or avant-garde. But it only works in dark mode, a version for light mode needs to be made.
GitHub allows you to change the image version according to the theme context:  [see here](https://github.blog/changelog/2022-05-19-specify-theme-context-for-images-in-markdown-beta/)

**Comment by Anirban166** [comment]
🙌  @astroesteban @ThibFrgsGmz Thank you Esteban and Thibault, I'm glad you both like it!

> The first two could be integrated into the GDS, the first in dark mode and the second in light mode.

By GDS, do you mean [this repository](https://github.com/fprime-community/fprime-gds) or perhaps the [website](https://nasa.github.io/fprime/UsersGuide/gds/gds-introduction.html)? (as a side note, I just see an old [issue](https://github.com/nasa/fprime/issues/287) that mentions replacing the screenshots there - that would be a good one to contribute and wrap up in the future, to keep everything up to date on the visuals)

> Regarding the ones with the wording, I prefer the second one which looks more futuristic or avant-garde. But it only works in dark mode, a version for light mode needs to be made.

The second one is the go-to :)
And yes, you're right. I did think about that last night but I was too sleepy to apply the Martian texture shading to the entire font in scale or to overlay it with a background of similar texture. I'll get to it soon later, hopefully before I catch my red-eye flight tonight ✈️ 

> GitHub allows you to change the image version according to the theme context: [see here](https://github.blog/changelog/2022-05-19-specify-theme-context-for-images-in-markdown-beta/)

Thanks for sharing that thread, I didn't know about that!

Until now, I went with either one of these two self-established design principles for logos that I make for use here and on my website - use a different colour than completely white/black (or even exact shades of grey that a GitHub theme renders, making logos of that colour appear invisible) or if I absolutely want to make it one of those, I apply a thin layer of glow to the boundaries of the letters/design in the logo with the opposite colour (if I want my logo text to be black, I do a white glow, for [e.g.](https://github.com/Anirban166/Zephyr/blob/main/Images/Finalized%20logo%20(reworked%20for%20greater%20width).png)) so as to negate the effect when a background with that colour comes into play, and to appear not too bad when the BG is composed of other colours.

**Reply by ThibFrgsGmz** [reply]
I was thinking about the graphical interface (WebApp)

**Comment by Anirban166** [comment]
Here you go:

<p align = "center">
<img width = "100%" height = "auto" src = "https://user-images.githubusercontent.com/30123691/186811255-4d9cb013-7689-4c27-bd56-cffc5a0b6569.png"> <be>   
<strong>A Flight-Proven, Multi-Platform, Open-Source Flight Software Framework</strong> <br>
</p>
<p align = "center">
<a href = "https://lgtm.com/projects/g/nasa/fprime/context:cpp"></a>
<img src = "https://img.shields.io/lgtm/grade/cpp/g/nasa/fprime.svg?logo=lgtm&logoWidth=18" alt = "Language grade: C++"> <a href = "https://lgtm.com/projects/g/nasa/fprime/context:python"></a> <img src = "https://img.shields.io/lgtm/grade/python/g/nasa/fprime.svg?logo=lgtm&logoWidth=18" alt = "Language grade: Python"> <br>
</p>

F´ (F Prime) is a component-driven framework that enables rapid development and deployment of spaceflight and other embedded software applications. Originally developed at the [Jet Propulsion Laboratory](https://www.jpl.nasa.gov/), F´ has been successfully deployed on several space applications. It is tailored but not limited to small-scale spaceflight systems such as CubeSats, SmallSats, and instruments.

...

<hr>

Let me know if this looks okay and if you guys have any suggestions for improvement or would like some particular change in the design! (also waiting for word from the members of the fprime team on this)
I'll be out till the weekend though (giving a talk at the HackCon X conference in Rock Hills, NY), so I'll check back next week.

**Reply by astroesteban** [reply]
Wow that's incredible! Thanks again for putting in the time to make this.
One minor comment on it is that it becomes a bit difficult to read in dark mode. Maybe it can be swapped out for the same font with a lighter color scheme in dark mode using that GitHub feature that @ThibFrgsGmz linked to? Thoughts?

<img width="855" alt="Screen Shot 2022-08-26 at 10 01 10 AM" src="https://user-images.githubusercontent.com/12473592/186934810-558f3cae-3a44-4e72-95ff-8d232fe8a766.png">

**Reply by ThibFrgsGmz** [reply]
I have the same feeling as @astroesteban, in dark mode there is not enough contrast. 😀

**Reply by Anirban166** [reply]
I agree too. For better contrast in dark mode, the colours would have to be lighter, something like:

<p align = "center">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://user-images.githubusercontent.com/30123691/187773809-9d3ea348-0a25-4ade-9268-c82aa62d637a.png">
  <img alt="Fprime logo: F' Flight Software and Embedded Systems Framework" src="https://user-images.githubusercontent.com/30123691/186811255-4d9cb013-7689-4c27-bd56-cffc5a0b6569.png">
</picture> <br> 
<strong>A Flight-Proven, Multi-Platform, Open-Source Flight Software Framework</strong> <br>
</p>
<p align = "center">
<a href = "https://lgtm.com/projects/g/nasa/fprime/context:cpp"></a>
<img src = "https://img.shields.io/lgtm/grade/cpp/g/nasa/fprime.svg?logo=lgtm&logoWidth=18" alt = "Language grade: C++"> <a href = "https://lgtm.com/projects/g/nasa/fprime/context:python"></a> <img src = "https://img.shields.io/lgtm/grade/python/g/nasa/fprime.svg?logo=lgtm&logoWidth=18" alt = "Language grade: Python"> <br>
</p>
<br>

(Edit: ^Using the code to switch in between image sources based on the GitHub color scheme in play)

Switching the mid-level shade with a tone closer to `#ffffff` than `#000000`  kinda does the job as I see. For example, if you compare the letters with the one for the light mode, you can see that I replaced the dark brown line that cuts through them in the middle with a much lighter version (more in comparison to the parts above and below that line) for this one.

**Reply by astroesteban** [reply]
I just changed to dark mode and that reads a lot better. Great work! 🚀

**Reply by Anirban166** [reply]
> I just changed to dark mode and that reads a lot better. Great work! 🚀

Thanks! 🙌

**Reply by ThibFrgsGmz** [reply]
Yeah better! Now we just have to wait for the feedback from more community members and original maintainers!╰(*°▽°*)╯
